kernel/src/main/java/org/kframework/unparser/ToLatex.java

```diff
@@ -0,0 +1,208 @@
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), def);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K Definition to LaTeX", e);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Definition def) throws IOException {
+        writeString(out, ("\\outerDefinition{"));
+        for (Module mod: JavaConverters.seqAsJavaList(def.modules().toSeq())) {
+            apply(out, mod);
+        }
+        writeString(out, "}{");
+        apply(out, def.att());
+        writeString(out, "}");
+    }
+
+    public static byte[] apply(Module mod) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), mod);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K Module to LaTeX", e);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Module mod) throws IOException {
+        writeString(out, ("\\outerModule{" + mod.name() + "}{"));
+        for (Module m: JavaConverters.seqAsJavaList(mod.imports().toSeq())) {
+            writeString(out, "\\outerImport{" + m.name() + "}");
+        }
+        writeString(out, "}{");
+        for (Sentence sent: JavaConverters.seqAsJavaList(mod.localSentences().toSeq())) {
+            apply(out, sent);
+        }
+        writeString(out, ("}{" + mod.att() + "}"));
+    }
+
+    public static byte[] apply(Sentence sent) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), sent);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K term to LaTeX", e, sent);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Sentence sent) throws IOException {
+        if (sent instanceof Rule) {
+            Rule rule = (Rule) sent;
+            writeString(out, "\\outerRule{");
+            apply(out, rule.body());
+            writeString(out, "}{");
+            apply(out, rule.requires());
+            writeString(out, "}{");
+            apply(out, rule.ensures());
+            writeString(out, "}{");
+            apply(out, rule.att());
+            writeString(out, "}");
+
+        } else if (sent instanceof ModuleComment) {
+            ModuleComment moduleComment = (ModuleComment) sent;
+            writeString(out, "\\outerModuleComment{" + moduleComment.comment() + "}{");
+            apply(out, moduleComment.att());
+            writeString(out, "}");
+
+        } else {
+            KEMException.criticalError("Do not know how to serialize sentence: " + sent.toString());
+        }
+    }
+
+    public static void apply(DataOutputStream out, K k) throws IOException {
```

Comment:
let's use a visitor for this.

Reply:
I modeled this one off of the Binary/JSON serializers, and would prefer to keep the same structure as those ones here (then if desired, refactor them all in a later PR to abstract out the common details).